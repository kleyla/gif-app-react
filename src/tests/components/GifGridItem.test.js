import { shallow } from "enzyme";
import { GifGridItem } from "./../../components/GifGridItem";

describe("Prueba en GifGridItem", () => {
  const title = "Hola";
  const url = "http://hi";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("Debe de mostrar el componente correctamente ", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de tener un parrafo con el title", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("Debe de tener la imagen igual al url y alt de los props", () => {
    const img = wrapper.find("img");
    // console.log(img.html());
    // console.log(img.props().src);
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("Debe de tener animate__fadeIn", () => {
    const div = wrapper.find("div");
    // console.log(div.prop('className'));
    const className = div.prop("className");

    expect(className.includes("animate__fadeIn")).toBe(true);
  });
});
