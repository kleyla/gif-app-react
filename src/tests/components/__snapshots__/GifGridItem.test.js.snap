// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en GifGridItem Debe de mostrar el componente correctamente  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Hola"
    src="http://hi"
  />
  <p>
    Hola
  </p>
</div>
`;
